src/core/types/keys/diffie_hellman/private_key.rs

```diff
@@ -0,0 +1,195 @@
+use crate::prelude::*;
+use crypto::keys::x25519::SecretKey as X25519PrivateKey;
+use hkdf::Hkdf;
+use k256::sha2::Sha256;
+
+#[derive(derive_more::Debug)]
+#[debug("{}", self.to_hex())]
+pub struct DiffieHellmanPrivateKey(X25519PrivateKey);
```

Comment:
@CyonAlexRDX Here I have added the DH Private/Public key pair, please advice if something should be different.

Reply:
Perhaps `KeyAgreementPrivateKey` is better than `DiffieHellman` ? Like CryptoKit uses.